js/models/Profile.js

```diff
@@ -0,0 +1,158 @@
+import _ from 'underscore';
+import BaseModel from './BaseModel';
+import app from '../app';
+import is from 'is_js';
+
+export default class extends BaseModel {
+  defaults() {
+    return {
+      primaryColor: '#086A9E',
+      secondaryColor: '#317DB8',
+      textColor: '#ffffff',
+      social: [],
+    };
+  }
+
+  url() {
+    // url is handled by sync, but backbone bombs if I don't have
+    // something explicitly set
+    return 'use-sync';
+  }
+
+  getColorFields() {
+    return [
+      'primaryColor',
+      'secondaryColor',
+      'textColor',
+    ];
+  }
+
+  // Ensure any colors are strings and have a leading hash.
+  standardizeColorFields(attrs = {}) {
+    const updatedAttrs = { ...attrs };
+
+    this.getColorFields().forEach((field) => {
+      if (typeof attrs[field] !== 'undefined') {
+        updatedAttrs[field] = updatedAttrs[field].toString();
+        updatedAttrs[field] = updatedAttrs[field].charAt(0) !== '#' ?
+          `#${updatedAttrs[field]}` : updatedAttrs[field];
+      }
+    });
+
+    return updatedAttrs;
+  }
+
+  set(key, val, options) {
+    // Handle both `"key", value` and `{key: value}` -style arguments.
+    let attrs;
+    let opts = options;
+
+    if (typeof key === 'object') {
+      attrs = key;
+      opts = val;
+    } else {
+      (attrs = {})[key] = val;
+    }
+
+    if (attrs.social) {
+      if (!_.isEqual(attrs.social, this._social)) {
+        this.attributes.social = this._social;
```

Comment:
Backbone uses it internally. One of it's uses is to compare against to determine if an attribute changed and consequentially if change events should be fired. It should usually not be directly updated (or very very judiciously if so). It was necessary to be updated in order for change event s to properly fire on the nested attribute.